### PR TITLE
refactor: Update proto with SDK API

### PR DESCRIPTION
**Pull request by mrrtree**
1. Add real `table_schema` from server in QueryResponse;
2. The `schema_content` will be deprecated after all  SDK upgrade

### CODE REVIEW

File: golang/pkg/ceresprompb/prometheus.pb.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.28.1
-// 	protoc        v3.20.1
+// 	protoc        v3.21.12
```
Comment (ZuLiangWang) [inline golang/pkg/ceresprompb/prometheus.pb.go:4]: Check you local `protoc` version, keep same as before.

File: protos/storage.proto [hunk 2]
```diff
@@ -105,7 +105,8 @@ message QueryResponse {
     JSON = 1;
   }
   SchemaType schema_type = 2;
-  string schema_content = 3;
+  string schema_content = 3; // versions after table_schema will be deprecated
```
Comment (chunshao90) [inline protos/storage.proto:108]: Remove `schema_type` and `schema_content`,  no need to consider compatibility.

**Comment by chunshao90** [comment]
Add a parameter to determine whether to return `tablesShema`?

File: golang/pkg/ceresprompb/prometheus.pb.go (hunk 1 above)
Comment (mrrtree) [reply golang/pkg/ceresprompb/prometheus.pb.go]: ok，done

File: protos/storage.proto (hunk 2 above)
Comment (mrrtree) [reply protos/storage.proto]: get.
rm `schema_content` already, and rename `schema_type` to `serialization_type` for avoid ambiguity

File: java/pom.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.ceresdb</groupId>
     <artifactId>ceresdb-proto-internal</artifactId>
-    <version>0.1.0</version>
+    <version>0.3.0</version>
```
Comment (archerny) [inline java/pom.xml:6]: We do not have 0.2.0. How about use 0.2.0 instead of 0.3.0

Comment (mrrtree) [reply java/pom.xml]: i make a mistake,  fix already

File: protos/storage.proto [hunk 4]
```diff
@@ -100,12 +100,5 @@ message QueryRequest {
...
   common.ResponseHeader header = 1;
-  enum SchemaType {
-    AVRO = 0;
-    JSON = 1;
-  }
-  SchemaType schema_type = 2;
-  string schema_content = 3;
-  repeated bytes rows = 4;
-  uint32 affected_rows = 5;
+  bytes arrow_bytes = 2;
```
[the remark this answers is not in the document]
Comment (mrrtree) [reply protos/storage.proto]: ok，done

File: protos/storage.proto [hunk 5]
```diff
@@ -94,18 +94,18 @@ message WriteResponse {
...
 
 message QueryResponse {
   common.ResponseHeader header = 1;
-  enum SchemaType {
-    AVRO = 0;
-    JSON = 1;
+  uint32 affected_rows = 2;
+  oneof rowsPayload {
+    ArrowPayload arrowPayload = 3;
   }
```
Comment (ShiKaiWi) [inline protos/storage.proto:106]: The style should be snake case:
```suggestion
  oneof rows_payload {
    ArrowRowsPayload payload = 3;
  }
```
And ArrowPayload seems not a good name, and I'm in favor of `ArrowRowsPayload`.

File: protos/storage.proto [hunk 6]
```diff
@@ -76,15 +76,15 @@ message WriteEntry {
   repeated FieldGroup field_groups = 2;
 }
 
-message WriteMetric {
-  string metric = 1;
+message WriteTable {
```
Comment (ShiKaiWi) [inline protos/storage.proto:79]: The name is weird, maybe we should call it `WriteTableRequest`?

File: protos/storage.proto [hunk 7]
```diff
@@ -76,15 +76,15 @@ message WriteEntry {
...
+  string table = 1;
   repeated string tag_names = 2;
   repeated string field_names = 3;
   repeated WriteEntry entries = 4;
 }
 
 message WriteRequest {
-  repeated WriteMetric metrics = 1;
+  repeated WriteTable tables = 1;
```
Comment (ShiKaiWi) [inline protos/storage.proto:87]: Maybe we should call it `table_requests` if we modify the `WriteTable` to `WriteTableRequet`.

Comment (mrrtree) [reply protos/storage.proto:87]: Shouldn't the protocol be included in the name? Like：
`pointWriteRequest` or `SqlRequest`

File: protos/storage.proto [hunk 8]
```diff
@@ -93,19 +93,19 @@ message WriteResponse {
...
+    ArrowPayload payload = 3;
   }
-  SchemaType schema_type = 2;
-  string schema_content = 3;
-  repeated bytes rows = 4;
-  uint32 affected_rows = 5;
 }
+
+message ArrowPayload {
+  bytes bytes = 1;
```
Comment (ShiKaiWi) [inline protos/storage.proto:110]: I guess this field type and name should be:
```suggestion
  repeated bytes record_batches = 1;
```

Comment (mrrtree) [reply protos/storage.proto]: ok, done

File: protos/storage.proto [hunk 9]
```diff
@@ -93,19 +93,19 @@ message WriteResponse {
   uint32 failed = 3;
 }
 
-message QueryRequest {
-  repeated string metrics = 1;
+message SqlQueryRequest {
+  repeated string tables = 1;
   string ql = 2;
```
Comment (Rachelint) [inline protos/storage.proto:98]: How about ql -> sql.

File: protos/storage.proto [hunk 10]
```diff
@@ -93,19 +93,24 @@ message WriteResponse {
...
 
-message QueryResponse {
+message SqlQueryResponse {
   common.ResponseHeader header = 1;
-  enum SchemaType {
-    AVRO = 0;
-    JSON = 1;
+  uint32 affected_rows = 2;
+  oneof rows_payload {
+    ArrowPayload payload = 3;
```
Comment (ShiKaiWi) [inline protos/storage.proto:105]:
```suggestion
    ArrowPayload arrow = 3;
```

File: protos/storage.proto [hunk 11]
```diff
@@ -93,19 +93,24 @@ message WriteResponse {
...
-message QueryResponse {
+message SqlQueryResponse {
   common.ResponseHeader header = 1;
-  enum SchemaType {
-    AVRO = 0;
-    JSON = 1;
+  uint32 affected_rows = 2;
+  oneof rows_payload {
+    ArrowPayload arrow = 3;
   }
```
Comment (ShiKaiWi) [inline protos/storage.proto:106]:
```suggestion
  oneof output {
    uint32 affected_rows = 2;
    ArrowPayload arrow = 3;
  }
```

**Review by chunshao90 — APPROVED** [review]
LGTM

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**